packages/contracts/test/voting/whitelist-voting.ts

```diff
@@ -169,7 +169,7 @@ describe('WhitelistVoting', function () {
       expect(await voting.canVote(0, ownerAddress)).to.equal(true);
       expect(await voting.canVote(0, user1)).to.equal(false);
 
-      expect(vote.actions).to.eql([
```

Comment:
This fixes and error that somehow was introduced by the commit I have branched off from (a3cb06037680808d64459bf3ac1ad5854c12cc5a). 
Without this, the tests fail and there is no reason to use `.eql` over `.equal`. However, this is strange and I can check this again.

Reply:
It is super strange. I reverted the changes (back to the usage of `eql` and I get the same error as before - also on the GitHub runner:
https://github.com/aragon/core/runs/7001078290?check_suite_focus=true

There is nothing happening in my PR that should affect the test (see the comparison 
https://github.com/aragon/core/compare/a3cb06037680808d64459bf3ac1ad5854c12cc5a...fix/deep-equal-error-in-voting).

**Update:**
I think I found a clue: 
We are overwriting the chai `equal` with a custom `equal` being "adapted from Waffle" 
https://github.com/aragon/core/blob/9030920927c7367265c6b9aa1082368bcf17301a/packages/contracts/test/test-utils/equal-overwrite.ts

@novaknole, do you understand what is happening here, and can you give us a hint?

Reply:
I would suggest keeping `deepEqual` as a separate function of our own, and use it only to explicitly assert on those places where deep equality is expected, keeping `equal` by default

Reply:
Is there anything speaking against simply using https://www.chaijs.com/api/bdd/#method_deep, which uses
https://github.com/chaijs/deep-eql @novaknole ?

Reply:
I see the problem, `.to.emit(...).withArgs(...)` uses the normal chai `equal` and can therefore not check events properly.

```js
expect(await dao.execute(0, dummyActions))
        .to.emit(dao, EVENTS.Executed)
        .withArgs(
          ownerAddress,
          0,
          [
            [
              dummyActions[0].to,
              ethers.BigNumber.from(dummyActions[0].value),
              dummyActions[0].data,
            ],
          ],
          expectedDummyResults
        );
    });
```

Reply:
Since we understand the problem now that happens with voting and fix it `equal` overwrite that @novaknole came up with, let's use `equal` instead of `eql` here and proceed. Do you agree @Rekard0 @brickpop ?

Reply:
i personally prefer to revert the change that cause this issue in the first place before, because i am afraid that might cause problem when another pr is merged that not have this problem,

unless there is a clear update of deps  ... etc

but i will leave the final decision to you guys

Reply:
I would personally avoid hidden or unexpected code that only becomes apparent when a weird error occurred. The time spent on figuring this out is an example of why I'd prefer straightforward and clearly visible tooling. 

@Michael-A-Heuer isn't it possible to execute the transaction, capture the events ourselves, and assert that at least one of them includes `Executed` with a deepEqual of the given arguments?

Reply:
I agree with all your concerns, but I can also clearly see the reason why Giorgi did this. 
Fixing this problem by removing Giorgis code touches at least 9 additional files and would pollute and delay this PR. 
I suggest that we create a separate JIRA issue for this that we can pull in the next sprint.

Reply:
We agreed together to open a [separate JIRA issue APP-616](https://aragonassociation.atlassian.net/browse/APP-616) and tackle it in the upcoming sprint.